src/NServiceBus.RabbitMQ/Connection/ConnectionFactory.cs

```diff
@@ -30,7 +30,8 @@ public ConnectionFactory(ConnectionConfiguration connectionConfiguration)
                 Password = connectionConfiguration.Password,
                 RequestedHeartbeat = connectionConfiguration.RequestedHeartbeat,
                 AutomaticRecoveryEnabled = true,
-                NetworkRecoveryInterval = connectionConfiguration.RetryDelay
+                NetworkRecoveryInterval = connectionConfiguration.RetryDelay,
+                UseBackgroundThreadsForIO = true,
```

Comment:
@bording I'm not sure if we needed to this or not in the 4.0.0 transport.

Was this part of fixing the hanging tests in 3.x or is it something else?